pkg/gardenlet/controller/managedseed/managedseed_actuator.go

```diff
@@ -0,0 +1,835 @@
+
+		if configValues, err = utils.SetMapToValues(configValues, gccValues, "gardenClientConnection"); err != nil {
+			return nil, err
+		}
+	}
+
+	// If a seed connection mechanism is specified, set seedClientConnection values accordingly
+	// Otherwise, if kubeconfig path is specified in seedClientConnection, read it and store its contents
+	var sccValues map[string]interface{}
+	if sccValues, err = utils.GetMapFromValues(configValues, "seedClientConnection"); err != nil {
+		return nil, err
+	}
+	if managedSeed.Spec.Gardenlet.SeedConnection != nil && *managedSeed.Spec.Gardenlet.SeedConnection == seedmanagementv1alpha1.SeedConnectionServiceAccount {
+		delete(sccValues, "kubeconfig")
+
+		if configValues, err = utils.SetMapToValues(configValues, sccValues, "seedClientConnection"); err != nil {
+			return nil, err
+		}
+	} else if kubeconfigPath, ok := sccValues["kubeconfig"].(string); ok && kubeconfigPath != "" {
+		sccValues = utils.InitValues(sccValues)
+		kubeconfig, err := ioutil.ReadFile(kubeconfigPath)
+		if err != nil {
+			return nil, err
+		}
+		sccValues["kubeconfig"] = string(kubeconfig)
+
+		if configValues, err = utils.SetMapToValues(configValues, sccValues, "seedClientConnection"); err != nil {
+			return nil, err
+		}
+	}
+
+	// Read TLS certificate and key files and store their contents
+	var tlsValues map[string]interface{}
+	if tlsValues, err = utils.GetMapFromValues(configValues, "server", "https", "tls"); err != nil {
+		return nil, err
+	}
+	if certPath, ok := tlsValues["serverCertPath"].(string); ok && certPath != "" && !strings.Contains(certPath, secrets.TemporaryDirectoryForSelfGeneratedTLSCertificatesPattern) {
+		tlsValues = utils.InitValues(tlsValues)
+		cert, err := ioutil.ReadFile(certPath)
+		if err != nil {
+			return nil, err
+		}
+		tlsValues["crt"] = string(cert)
+	}
+	delete(tlsValues, "serverCertPath")
+	if keyPath, ok := tlsValues["serverKeyPath"].(string); ok && keyPath != "" && !strings.Contains(keyPath, secrets.TemporaryDirectoryForSelfGeneratedTLSCertificatesPattern) {
+		tlsValues = utils.InitValues(tlsValues)
+		key, err := ioutil.ReadFile(keyPath)
+		if err != nil {
+			return nil, err
+		}
+		tlsValues["key"] = string(key)
+	}
+	delete(tlsValues, "serverKeyPath")
+	if configValues, err = utils.SetMapToValues(configValues, tlsValues, "server", "https", "tls"); err != nil {
+		return nil, err
+	}
+
+	// Prepare seed config, set seedConfig, and unset seedSelector
+	var seedConfig *configv1alpha1.SeedConfig
+	if configObj != nil {
+		seedConfig = configObj.SeedConfig
+	}
+	seedConfig, err = a.prepareSeedConfig(ctx, managedSeed, shoot, seedConfig)
+	if err != nil {
+		return nil, err
+	}
+	var seedConfigValues map[string]interface{}
+	if seedConfigValues, err = utils.ToValues(seedConfig); err != nil {
+		return nil, err
+	}
+	if configValues, err = utils.SetMapToValues(configValues, seedConfigValues, "seedConfig"); err != nil {
+		return nil, err
+	}
+	delete(configValues, "seedSelector")
```

Comment:
Only required when the parent configuration is merged? Otherwise specifying a SeedSelector should be forbidden during validation of the resource.

Reply:
Yes, but even in this case, we still need to explicitly delete it here to make sure we didn't inherit something wrong.

Reply:
> we didn't inherit something wrong.

that's what I meant with "Only required when the parent configuration is merged"

The problem that I see here again is, that we possibly delete something from the config that can either be inherited or an explicit user setting. 

If we want to forbid ManagedSeeds - GardenletConfiguration to contain a seedSelector, then we need a validation + only deletion when we actually merge the parent config. 
If we want to allow it (I can imagine that there are use cases where you want to deploy a Gardenlet for multiple seed clusters), then we cannot just always delete the seedSelector.

Reply:
> I think we just need a validation that the configuration does not contain a seedSelector upon ManagedSeed creation + only deletion of the selector when we actually merge the parent config.

I agree regarding the validation, but I don't see what's wrong with the current implementation. From my perspective, cleaning `seedSelector` here would anyway only clean anything if it was set in the parent. I could add an explicit check if merging with parent is enabled but I see no value in this - without it it's simpler, safer, and (assuming we have the validation) there is nothing that the user would regard as unexpected behavior.

Reply:
> I could add an explicit check if merging with parent is enabled but I see no value in this - without it it's simpler, safer, and (assuming we have the validation) there is nothing that the user would regard as unexpected behavior.

okay, valid point. Thanks for following up